src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4754,6 +4754,14 @@ private unsafe char[] GetCanonicalPath(char[] dest, ref int pos, UriFormat forma
                         end += (_info.Offset.Query - _info.Offset.Path);
                     }
                 }
+
+                // On Unix, escape '\\' in path of file uris to '%5C' canonical form.
+                if (!IsWindowsSystem && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes) && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && !IsImplicitFile)
```

Comment:
This code doesn't run on windows, instead of ifdef it uses conditional compilation of source files as done [here](https://github.com/dotnet/corefx/blob/master/src/System.Private.Uri/src/System.Private.Uri.csproj#L48), on windows IsWindowsSystem = true, so this if loop is not entered. This conditional compilation of source file is used in multiple places in this file, this project doesn't use the ifdef pattern.